### PR TITLE
Trivy vulnerability scanning

### PR DESCRIPTION
**Pull request by hamistao**
This adds a new workflow that uses Trivy to scan microcluster's dependencies daily.

### CODE REVIEW

**Comment by hamistao** [comment]
@tomponline @simondeziel This is also ready

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          format: "sarif"
+          output: "trivy-microcluster-repo-scan-results.sarif"
+          severity: "LOW,MEDIUM,HIGH,CRITICAL"
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v3
+        with:
+          sarif_file: "trivy-microcluster-repo-scan-results.sarif"
+          sha: ${{ github.sha }}
+          ref: refs/heads/main
```
Comment (simondeziel) [inline .github/workflows/security.yml]: Should we avoid hardcoding `main` here?

Comment (hamistao) [reply .github/workflows/security.yml]: would you suggest using a matrix strategy instead?

Comment (simondeziel) [reply .github/workflows/security.yml]: I just checked and there is no `main` branch, just `v2` and `v3` ATM so yeah, a matrix sounds like a good idea.

Comment (hamistao) [reply .github/workflows/security.yml]: Indeed, my fork does only have a main branch for some reason so this is why my tests worked, thanks for the warning!

Comment (hamistao) [reply .github/workflows/security.yml]: Done!

**Review by simondeziel — APPROVED** [review]

**Review by tomponline — APPROVED** [review]

**Pull request merged**